mtags/src/main/scala-3/scala/meta/internal/pc/MetalsInteractive.scala

```diff
@@ -149,4 +163,128 @@ object MetalsInteractive:
       case gtree: Select if isForComprehensionSyntheticName(gtree) => true
       case _ => false
 
+  @tailrec
+  def enclosingSymbols(
+      path: List[Tree],
+      pos: SourcePosition,
+      indexed: IndexedContext,
+      skipCheckOnName: Boolean = false
+  ): List[Symbol] =
+    import indexed.ctx
+    path match
+      // For a named arg, find the target `DefDef` and jump to the param
+      case NamedArg(name, _) :: Apply(fn, _) :: _ =>
+        val funSym = fn.symbol
+        if funSym.is(Synthetic) && funSym.owner.is(CaseClass) then
+          List(funSym.owner.info.member(name).symbol)
+        else
+          val classTree = funSym.topLevelClass.asClass.rootTree
+          val paramSymbol =
+            for
+              DefDef(_, paramss, _, _) <- tpd
+                .defPath(funSym, classTree)
+                .lastOption
+              param <- paramss.flatten.find(_.name == name)
+            yield param.symbol
+          List(paramSymbol.getOrElse(fn.symbol))
+
+      case (_: untpd.ImportSelector) :: (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      case (imp: Import) :: _ =>
+        importedSymbols(imp, _.span.contains(pos.span))
+
+      // wildcard param
+      case head :: _ if (head.symbol.is(Param) && head.symbol.is(Synthetic)) =>
+        List(head.symbol)
+
+      case (head @ Select(target, name)) :: _
+          if head.symbol.is(Synthetic) && name == StdNames.nme.apply =>
+        val sym = target.symbol
+        if sym.is(Synthetic) && sym.is(Module) then List(sym.companionClass)
+        else List(target.symbol)
+
+      // L@@ft(...)
+      case (head @ ApplySelect(select)) :: _
+          if (select.qualifier.sourcePos.contains(pos) &&
+            select.name == StdNames.nme.apply) =>
+        List(head.symbol)
+
+      // for comprehension
+      case (head @ ApplySelect(select)) :: _ if isForSynthetic(head) =>
+        // If the cursor is on the qualifier, return the symbol for it
+        // `for { x <- List(1).head@@Option }`  returns the symbol of `headOption`
+        if select.qualifier.sourcePos.contains(pos) then
+          List(select.qualifier.symbol)
+        // Otherwise, returns the symbol of for synthetics such as "withFilter"
+        else List(head.symbol)
```

Comment:
Just a question.
What do we receive in path as a head in `for { x <- List(1).head@@Option }` ?

I'm wondering if there is an issue with `Interactive.pathTo`
Looks like it should be just a select but why this case doesn't work?

Reply:
In this case, `path.head` is something like

```scala
Apply(
  TypeApply(
    Select(
      Select(
        ...
        headOption
      ),
      map
    ),
    List(...)
  ),
  List(...)
  )
)
```

Maybe this is an issue of `Interacitve.pathTo` which should return the path with the head like:

```scala
Select(
  ...
  headOption
),
```

Reply:
thx! Looks like it is. But I'm fine the your fix. We can investigate the issue with `Interacitve.pathTo` later